### PR TITLE
mcelog: Bug fixes and code style changes

### PR DESCRIPTION
**Pull request by pszczerx**
This series of patches introduces following changes:

- Increase delay between socket re-initialization from 1ms to 1000ms
- Check if location (plugin instance) was correctly parsed
- Dispatch notification on successful connection to mcelog server
- Free notification metadata
- Other minor changes

### CODE REVIEW

**Comment by rubenk** [comment]
This seems to fix several things. Could you split it up into multiple commits, one commit per change?

**Comment by pszczerx** [comment]
@rubenk wouldn't it be enough to just modify the commit title to better reflect the changes? It's quite a small commit and I'm not sure if it needs to be split up. I have already squashed the changes on my local branch, so splitting the commit would require more effort on my side. If you are fine with such approach, do you have a preferred title?

**Comment by rubenk** [comment]
I wouldn't call a 150 line diff a small commit. I'm sorry, but squashing everything together makes reviewing very hard.

**Comment by pszczerx** [comment]
@rubenk, as requested, commit has been split.

File: src/mcelog.c [hunk 1]
```diff
@@ -210,95 +213,105 @@ static int socket_reinit(socket_adapter_t *self) {
...
 
-static int mcelog_submit(mcelog_memory_rec_t mr) {
+static int mcelog_submit(const mcelog_memory_rec_t *mr) {
 
-  value_list_t vl = VALUE_LIST_INIT;
-  vl.values_len = 1;
-  vl.time = cdtime();
+  if (!mr) {
+    ERROR(MCELOG_PLUGIN ": %s: NULL pointer", __FUNCTION__);
+    return -1;
```
Comment (rubenk) [inline src/mcelog.c]: return (-1)

File: src/mcelog.c [hunk 2]
```diff
@@ -210,95 +213,105 @@ static int socket_reinit(socket_adapter_t *self) {
...
-    ERROR("%s: add corrected errors period meta data failed", MCELOG_PLUGIN);
+  if (plugin_notification_meta_add_signed_int(n,
+                                              "uncorrected memory timed errors",
+                                              mr->uncorrected_err_timed) < 0) {
+    ERROR(MCELOG_PLUGIN ": add corrected timed errors meta data failed");
     plugin_notification_meta_free(n->meta);
     return (-1);
   }
+  if (mr->uncorrected_err_timed_period[0] != '\0')
+    if (plugin_notification_meta_add_string(n, "uncorrected errors time period",
```
Comment (rubenk) [inline src/mcelog.c]: If you reverse the condition and return early, you can get rid of the nested indentation. Alternatively, combine the two conditions and && them into a single if.

File: src/mcelog.c [hunk 3]
```diff
@@ -210,95 +213,105 @@ static int socket_reinit(socket_adapter_t *self) {
...
 
   ssnprintf(vl.type_instance, sizeof(vl.type_instance),
-            "uncorrected_memory_errors_in_%s", mr.uncorrected_err_timed_period);
-  vl.values = &(value_t){.derive = (derive_t)mr.uncorrected_err_timed};
+            "uncorrected_memory_errors_in_%s",
+            mr->uncorrected_err_timed_period);
+  vl.values = &(value_t){.derive = (derive_t)mr->uncorrected_err_timed};
   plugin_dispatch_values(&vl);
 
   return 0;
```
Comment (rubenk) [inline src/mcelog.c]: unrelated to this PR, but I just now noticed this. This file mixes return n and return (n). Could you pick one style?

**Pull request merged**